templates/shared_services/nexus-cert/terraform/appgateway.tf

```diff
@@ -0,0 +1,171 @@
+resource "azurerm_public_ip" "appgwpip" {
+  name                = "pip-nexus-${var.tre_id}"
+  resource_group_name = local.core_resource_group_name
+  location            = var.location
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  domain_name_label   = "nexus-${var.tre_id}"
+
+  lifecycle { ignore_changes = [tags] }
+}
+
+resource "azurerm_user_assigned_identity" "agw_id" {
+  resource_group_name = local.core_resource_group_name
+  location            = var.location
+  name                = "id-agw-nexuscert-${var.tre_id}"
+
+  lifecycle { ignore_changes = [tags] }
+}
+
+resource "azurerm_application_gateway" "agw" {
```

Comment:
Are we creating a new AppGateway?

Reply:
Currently, yes. But I imagine a single shared gateway may be adaquate.  But I will test this fully on Monday as I ran out of time today.

I had designed it as a separate component for now specifically for letsencrypt but if integrated I`m happy to ammended.

Reply:
@ross-p-smith After a bit of digging I don't think we'd be able to amend the existing app gateway on deployment of this and a new one would be required.  It wouldn't be needed outside of acquiring the certificate however so could be stopped/started when required which would avoid the additional cost?

Reply:
I have also tested the uses of a container instance although this is resolving to the wrong FQDN.  The IP seems to be generated by the CI so ends in azurecontainer.io.  This would maybe work if Nexus was also hosted in a CI although would then need to look at writing the auth hook to the container rather than a storage account blob.